### PR TITLE
Add library for render Markdown in HTML

### PR DESCRIPTION
**Pull request by sahsisunny**
### Closes - #572 

**Description:**

I have added a *markdown-to-jsx* package to convert the markdown content to JSX. This package converts the markdown content to JSX and then renders it on the page. For this, I have created a new component called *MarkdownRenderer* which takes the markdown content as a prop and converts it to JSX using the *markdown-to-jsx* package. And I also added the user and assignee `html_url` to the issue card so that the user can directly go to the user's or assignee's profiles by clicking on the link.

**Screenshot**
**Before:**
Screenshot (175)

**After:**

https://github.com/Real-Dev-Squad/website-status/assets/70854507/04f10b15-9175-47cd-bd7f-be8da0f97b71

### CODE REVIEW

File: __tests__/Unit/Components/Issues/Card.test.tsx [hunk 1]
```diff
@@ -8,11 +8,13 @@ const DEFAULT_PROPS: IssueItem = {
     title: 'Status view to all features being built for our app',
     user: {
         login: 'ankushdharkar',
+        html_url: 'https://github.com/ankushdharkar',
```
Comment (RitikJaiswal75) [inline __tests__/Unit/Components/Issues/Card.test.tsx:11]: please use dummy names

File: src/components/MarkdownRenderer/MarkdownRenderer.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import Markdown from 'markdown-to-jsx';
+
+interface MarkdownRendererProps {
+    content: string;
+    className?: string;
+}
+
+const MarkdownRenderer: React.FC<MarkdownRendererProps> = ({
+    content,
+    className,
```
Comment (RitikJaiswal75) [inline src/components/MarkdownRenderer/MarkdownRenderer.tsx:11]: why are we passing style/class name as a prop?

File: src/components/issues/Card.module.scss [hunk 3]
```diff
@@ -11,7 +11,7 @@ $card-text: #78716c;
     transition: 0.2s ease-out;
     display: flex;
     flex-direction: column;
-    margin: 2rem auto;
+    margin: 1rem auto;
```
Comment (RitikJaiswal75) [inline src/components/issues/Card.module.scss:14]: please use even numbers, also why was this changed?

File: src/components/issues/Card.tsx [hunk 4]
```diff
@@ -107,6 +74,53 @@ const Card: FC<IssueCardProps> = ({ issue }) => {
                     Convert to task
                 </button>
             </div>
+            <p className={styles.card__issue_created__by}>
+                Opened on {created} by
```
Comment (RitikJaiswal75) [inline src/components/issues/Card.tsx:78]: can we use `date` instead of created?

File: src/components/issues/Card.tsx [hunk 5]
```diff
@@ -107,6 +74,53 @@ const Card: FC<IssueCardProps> = ({ issue }) => {
...
+                    target="_blank"
+                    rel="noreferrer"
+                >
+                    {issue.user.login}
+                </a>
+            </p>
+            <div className="card__body">
+                <MarkdownRenderer
+                    className={styles.card__body}
+                    content={issue.body ?? 'No description provided.'}
```
Comment (RitikJaiswal75) [inline src/components/issues/Card.tsx:90]: did you test for this, when no content is provided? I cant see it in the test file.

File: src/components/issues/Card.tsx [hunk 6]
```diff
@@ -107,6 +74,53 @@ const Card: FC<IssueCardProps> = ({ issue }) => {
...
+            </div>
+            {issue.assignee && (
+                <p className={styles.card__assignee}>
+                    Assigned to:
+                    <a
+                        href={issue.assignee?.html_url ?? '#'}
+                        target="_blank"
+                        rel="noreferrer"
+                    >
+                        {issue.assignee?.login}
```
Comment (RitikJaiswal75) [inline src/components/issues/Card.tsx:101]: i cant see test for this as well if the assignee exists or not

File: src/components/MarkdownRenderer/MarkdownRenderer.tsx (hunk 2 above)
Comment (sahsisunny) [reply src/components/MarkdownRenderer/MarkdownRenderer.tsx:11]: For Customization and Reusability.

File: src/components/issues/Card.module.scss (hunk 3 above)
Comment (sahsisunny) [reply src/components/issues/Card.module.scss:14]: Why use Even not Odd? For styling purpose.

File: __tests__/Unit/Components/Issues/Card.test.tsx (hunk 1 above)
Comment (sahsisunny) [reply __tests__/Unit/Components/Issues/Card.test.tsx:11]: It is necessary? btw it was already written.

File: src/components/issues/Card.tsx (hunk 6 above)
Comment (sahsisunny) [reply src/components/issues/Card.tsx:101]: See this - https://github.com/Real-Dev-Squad/website-status/pull/580/files#diff-d45c985390f469d6629dabf9546686d836be99a3b27bf0a7a41a48f3e32bf8c0R42-R51

File: src/components/issues/Card.module.scss [hunk 7]
```diff
@@ -22,70 +22,116 @@ $card-text: #78716c;
...
+    table {
+        width: 100%;
+        border-collapse: collapse;
+        border: 1px solid $card-text;
+        border-radius: 0.5rem;
+        margin: 0.5rem 0;
+        th,
+        td {
+            border: 1px solid $card-text;
+            padding: 0.5rem;
```
Comment (Pratiyushkumar) [inline src/components/issues/Card.module.scss:124]: use rem instead of px and there are many places where your have used px convert it into rem or em

File: src/components/issues/Card.tsx [hunk 8]
```diff
@@ -107,6 +74,53 @@ const Card: FC<IssueCardProps> = ({ issue }) => {
                     Convert to task
                 </button>
```
Comment (Pratiyushkumar) [inline src/components/issues/Card.tsx:75]: button type required

File: src/components/issues/Card.tsx (hunk 4 above)
Comment (Pratiyushkumar) [reply src/components/issues/Card.tsx:78]: please put clear variable name that is understandable

**Review by Pratiyushkumar — CHANGES_REQUESTED** [review]
@sahsisunny  please try to resolve the comments that have already been done on your PR.

File: src/components/issues/Card.module.scss (hunk 7 above)
Comment (sahsisunny) [reply src/components/issues/Card.module.scss:124]: Where?

File: src/components/issues/Card.tsx (hunk 8 above)
Comment (sahsisunny) [reply src/components/issues/Card.tsx:75]: Who created this button?

File: src/components/MarkdownRenderer/MarkdownRenderer.tsx (hunk 2 above)
Comment (RitikJaiswal75) [reply src/components/MarkdownRenderer/MarkdownRenderer.tsx:11]: how does passing classname increase reusability, also where else do we plan to use this?

Comment (sahsisunny) [reply src/components/MarkdownRenderer/MarkdownRenderer.tsx:11]: When we pass the class name we can set the custom style and we can use it in other places like showing the pull requests etc

Comment (sahsisunny) [reply src/components/MarkdownRenderer/MarkdownRenderer.tsx:11]: Can you tell me the issue when I am passing the class name?

File: src/components/issues/Card.module.scss (hunk 3 above)
Comment (sahsisunny) [reply src/components/issues/Card.module.scss:14]: Can you provide me with the docs for it?

File: src/components/issues/Card.tsx (hunk 4 above)
Comment (sahsisunny) [reply src/components/issues/Card.tsx]: Done

File: src/components/issues/Card.tsx (hunk 5 above)
Comment (sahsisunny) [reply src/components/issues/Card.tsx:90]: Done

File: __tests__/Unit/Components/Issues/Card.test.tsx (hunk 1 above)
Comment (RitikJaiswal75) [reply __tests__/Unit/Components/Issues/Card.test.tsx:12]: It is a good practice. Later on, someone can take action (even legally) against us for using their names without written consent.

File: src/components/MarkdownRenderer/MarkdownRenderer.tsx (hunk 2 above)
Comment (RitikJaiswal75) [reply src/components/MarkdownRenderer/MarkdownRenderer.tsx:11]: So when we pass the class name There are too many things to concern ourselves with, it would be easier for developers if they call on the markdown tag and just pass in content and it automatically renders it. As remembering class names will be hard.

**Review by shubhamsigdar1 — COMMENTED** [review]
great work @sahsisunny

File: __tests__/Unit/Components/Issues/Card.test.tsx [hunk 9]
```diff
@@ -8,26 +9,59 @@ const DEFAULT_PROPS: IssueItem = {
...
+
+    test('Should render the MarkdownRenderer component with the correct content', () => {
+        const screen = render(
+            <MarkdownRenderer
+                content={DEFAULT_PROPS.body ?? 'No description provided.'}
+            />
+        );
+
+        const contentElement = screen.getByText(
+            DEFAULT_PROPS.body ?? 'No description provided.'
```
Comment (shubhamsigdar1) [inline __tests__/Unit/Components/Issues/Card.test.tsx:62]: If I am not wrong sunny you deleted the dummy body 
Can you check for markdown content??

**Review by shubhamsigdar1 — CHANGES_REQUESTED** [review]

File: __tests__/Unit/Components/Issues/Card.test.tsx (hunk 9 above)
Comment (shubhamsigdar1) [inline __tests__/Unit/Components/Issues/Card.test.tsx:62]: If I am not wrong sunny you deleted the dummy body
Can you check for markdown content??

Comment (sahsisunny) [reply __tests__/Unit/Components/Issues/Card.test.tsx]: Yes, added another dummy issue with body for test

**Review by Pratiyushkumar — DISMISSED** [review]
Looks good to me

File: __tests__/Unit/Components/Issues/Card.test.tsx [hunk 10]
```diff
@@ -1,33 +1,85 @@
...
+        html_url: 'https://github.com/dummyuser',
+    },
+    labels: [],
+    state: 'open',
+    assignee: {
+        login: 'dummyassignee',
+        html_url: 'https://github.com/dummyassignee',
+    },
+    created_at: '2020-10-30T02:08:48Z',
+    body: 'Dummy issue description',
```
Comment (shubhamsigdar1) [inline __tests__/Unit/Components/Issues/Card.test.tsx:38]: Sunny your pr focused on convert markdown content into JSX....can you please add value markdown content in body for testing

Comment (sahsisunny) [reply __tests__/Unit/Components/Issues/Card.test.tsx:38]: It is the simple markdown content.

File: __tests__/Unit/Components/Issues/Card.test.tsx (hunk 1 above)
Comment (sahsisunny) [reply __tests__/Unit/Components/Issues/Card.test.tsx]: Done

File: __tests__/Unit/Components/Issues/Card.test.tsx (hunk 10 above)
Comment (shubhamsigdar1) [reply __tests__/Unit/Components/Issues/Card.test.tsx:38]: Roadmap\r\n',

Add this :)

File: src/components/MarkdownRenderer/MarkdownRenderer.tsx (hunk 2 above)
Comment (sahsisunny) [reply src/components/MarkdownRenderer/MarkdownRenderer.tsx:11]: I don't think it creates any issues because we are using these components for different purposes currently we are using for two purposes The first one is for the title and the second one is for the body of we pass the default class name so we can't customize the style of you want to remove the class name from propa then we have to create the two markdown renderer components.  What do you think ?

File: __tests__/Unit/Components/Issues/Card.test.tsx (hunk 10 above)
Comment (sahsisunny) [reply __tests__/Unit/Components/Issues/Card.test.tsx:38]: ?

Comment (RitikJaiswal75) [inline __tests__/Unit/Components/Issues/Card.test.tsx:38]: can we move these to a seperate file like fixtures/mocks?

File: __tests__/Unit/Components/Issues/Card.test.tsx [hunk 11]
```diff
@@ -1,33 +1,85 @@
...
-        const date = new Date(DEFAULT_PROPS.created_at).toDateString();
+        const screen = render(<Card issue={issue_one} />);
 
-        expect(screen.getByText(DEFAULT_PROPS.title)).toBeInTheDocument();
-        expect(screen.getByText(DEFAULT_PROPS.html_url)).toBeInTheDocument();
-        expect(
-            screen.getByText(`Opened on ${date} by ${DEFAULT_PROPS.user.login}`)
-        ).toBeInTheDocument();
+        expect(screen.getByText(issue_one.title)).toBeInTheDocument();
+        expect(screen.getByText(issue_one.html_url)).toBeInTheDocument();
```
Comment (RitikJaiswal75) [inline __tests__/Unit/Components/Issues/Card.test.tsx:46]: these texts just make sure if the card is rendered. Is there any way to check if they have rendered with the markdown?

File: __tests__/Unit/Components/Issues/Card.test.tsx (hunk 10 above)
Comment (sahsisunny) [reply __tests__/Unit/Components/Issues/Card.test.tsx]: done

File: __tests__/Unit/Components/Issues/Card.test.tsx (hunk 11 above)
Comment (sahsisunny) [reply __tests__/Unit/Components/Issues/Card.test.tsx]: done

**Review by RitikJaiswal75 — APPROVED** [review]
LGTM.

Please take other's approval as well

**Review by shubhamsigdar1 — APPROVED** [review]

**Review by Pratiyushkumar — APPROVED** [review]
looks good to me
